### PR TITLE
Add logging context for request including request id

### PR DESCRIPTION
**Pull request by coltonlw**
This adds a custom request object which has additional attributes "id" and "logger". logger is a LoggerAdapter which adds the request id to each message string. In the future this can be expanded to include additional per-request context information with each logging call, including extra keys when we log in JSON.  Also added a uwsgi log var for request_id so we can use that in our uwsgi logformat allowing us to match a request_id from python log messages to a message in the uwsgi access log.  uwsgi log var is optional so other webservers may be used

Example output:

```
2016-09-08 13:34:04  scitran.api  base.py  104:INFO Initialized request request_id=c88e9c98-1473359644
127.0.0.1 - - [08/Sep/2016:13:34:04 -0500] "GET /api/rules?root=true HTTP/1.1" 200 112 "-" "-" request_id=c88e9c98-1473359644
```

Fixes: #199, #451

### CODE REVIEW

**Comment by gsfr** [comment]
I'm good to go with this approach per offline discussion, but @nagem still needs to sign off on both approach and implementation.

File: api/api.py [hunk 1]
```diff
@@ -21,9 +22,15 @@
 from .handlers import searchhandler
 from .handlers import schemahandler
 from .handlers import reporthandler
+from .request import SciTranRequest
 
 log = config.log
```
Comment (nagem) [inline api/api.py]: Looks like you forgot one here.

**Comment by nagem** [comment]
I like it, looks good!

File: api/api.py (hunk 1 above)
Comment (coltonlw) [inline api/api.py]: Apparently we do use this logger in the new relic setup code

**Pull request merged**